iothub_client/src/iothub_client_ll.c

```diff
@@ -2114,3 +2152,30 @@ IOTHUB_CLIENT_RESULT IoTHubClient_LL_UploadMultipleBlocksToBlob(IOTHUB_CLIENT_LL
 }
 
 #endif /* DONT_USE_UPLOADTOBLOB */
+
+IOTHUB_CLIENT_RESULT IoTHubClient_LL_EnableDiagnostic(IOTHUB_CLIENT_LL_HANDLE iotHubClientHandle, int samplingPercentage)
+{
+    IOTHUB_CLIENT_LL_HANDLE_DATA* handleData = (IOTHUB_CLIENT_LL_HANDLE_DATA*)iotHubClientHandle;
+
+    if (samplingPercentage == USE_SERVER_DIAG_SETTINGS)
```

Comment:
I prefer to use another dedicated function to enable remote diag setting.

Reply:
Wait for new design doc